edge-modules/metrics-collector/src/Program.cs

```diff
@@ -45,15 +45,27 @@ static async Task<int> MainAsync()
 #endif
 
             LoggerUtil.Writer.LogInformation($"Metrics collector initialized with the following settings:\r\n{Settings.Information}");
-            var transportSetting = new AmqpTransportSettings(TransportType.Amqp_Tcp_Only);
-
-            ITransportSettings[] transportSettings = { transportSetting };
             ModuleClientWrapper moduleClientWrapper = null;
             try
             {
-                moduleClientWrapper = await ModuleClientWrapper.BuildModuleClientWrapperAsync(transportSettings);
-
-                PeriodicTask periodicIothubConnect = new PeriodicTask(moduleClientWrapper.RecreateClientAsync, Settings.Current.IotHubConnectFrequency, TimeSpan.FromMinutes(1), LoggerUtil.Writer, "Reconnect to IoT Hub", true);
+                try
+                {
+                    moduleClientWrapper = await ModuleClientWrapper.BuildModuleClientWrapperAsync(Settings.Current.TransportType);
+                    PeriodicTask periodicIothubConnect = new PeriodicTask(moduleClientWrapper.RecreateClientAsync, Settings.Current.IotHubConnectFrequency, TimeSpan.FromMinutes(1), LoggerUtil.Writer, "Reconnect to IoT Hub", true);
+                }
+                catch (Exception e)
```

Comment:
Are there exceptions that _aren't_ related to connecting to Edge Hub, that should cause the module to exit here, even if (UploadTarget == IotMessage)?

Reply:
I assume you mean "even if (UploadTarget == AzureMonitor)". For upload target of IotMessage we will always fail fast if a problem in this try block.

My logic was that whatever went wrong had to be either the module client creation or the periodic task creation, both of which apply only for the client which isn't necessary to be used for AzureMonitor upload path. So I think that there won't be any reason to exit and fail fast.

Reply:
> I assume you mean "even if (UploadTarget == AzureMonitor)"

Yes. 😏

> My logic was

Got it...